transmogrifier/sources/transformer.py

```diff
@@ -55,73 +68,246 @@ def __next__(self) -> TimdexRecord:
+
+        Must be overridden by source subclasses.
+
+        Args:
+            source_record: A single source record.
+        """
+        return ""
+
+    @classmethod
     @abstractmethod
-    def get_optional_fields(self, xml: Tag) -> Optional[dict]:
+    def record_is_deleted(cls, source_record: JSON | Tag) -> bool:
+        """
+        Determine whether record has a status of deleted.
+
+        Must be overridden by source subclasses.
+
+        Args:
+            source_record: A single source record.
+        """
+        return False
+
+    @abstractmethod
+    def get_required_fields(self, source_record: JSON | Tag) -> dict:
+        """
+        Get required TIMDEX fields from a source record.
+
+        Must be overridden by format subclasses.
+
+        Args:
+            source_record: A single source record.
+        """
+        return {}
+
+    @abstractmethod
+    def transform(self, source_record: JSON | Tag) -> Optional[TimdexRecord]:
+        """
+        Transform a source record into a TIMDEX record.
+
+        Must be overridden by format subclasses.
+
+        Args:
+            source_record: A single source record.
+        """
+        return None
+
+    @classmethod
+    @abstractmethod
+    def get_valid_title(cls, source_record_id: str, source_record: JSON | Tag) -> str:
+        """
+        Retrieves main title(s) from a source record and returns a valid title string.
+
+        Must be overridden by source subclasses.
+
+        Args:
+            source_record_id: Record identifier for the source record.
+            source_record: A single source record.
+        """
+        return ""
+
+    @classmethod
+    @abstractmethod
+    def get_source_link(
+        cls, source_base_url: str, source_record_id: str, source_record: JSON | Tag
+    ) -> str:
+        """
+        Class method to set the source link for the item.
+
+        Must be overridden by source subclasses.
+
+        Args:
+            source_base_url: Source base URL.
+            source_record_id: Record identifier for the source record.
+            source_record: A single source record.
+        """
+        return ""
+
+    @classmethod
+    @abstractmethod
+    def get_timdex_record_id(
+        cls, source: str, source_record_id: str, source_record: Tag
+    ) -> str:
+        """
+        Class method to set the TIMDEX record id.
+
+        Must be overridden by source subclasses.
+
+        Args:
+            source: Source name.
+            source_record_id: Record identifier for the source record.
+            source_record: A single source record.
+        """
+        return ""
+
+
+class XmlTransformer(Transformer):
+    """Base transformer class."""
+
+    @final
+    @classmethod
+    def parse_source_file(cls, input_file: str) -> Iterator[Tag]:
+        """
+        Parse source file and return source records via an iterator.
+
+        May not be overridden.
+        """
```

Comment:
I would suggest updating this docstring to be XML specific, thereby differentiating from the base `Transformer`.